dglgo/dglgo/pipeline/graphpred/graphpred.jinja-py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+import torch
+import torch.nn as nn
+from dgl.data import AsGraphPredDataset
+from dgl.dataloading import GraphDataLoader
+from sklearn.metrics import roc_auc_score
+from tqdm import tqdm
+{{ data_import_code }}
+
+{{ model_code }}
+
+def train(device, loader, model, criterion, optimizer):
+    model.train()
+
+    for _, (g, labels) in enumerate(tqdm(loader, desc="Iteration")):
+        if g.batch_size == 1:
+            continue
+
+        g = g.to(device)
+        labels = labels.to(device)
+        node_feat = g.ndata['feat']
+        edge_feat = g.edata['feat']
+
+        pred = model(g, node_feat, edge_feat)
+        optimizer.zero_grad()
+        # ignore nan targets (unlabeled) when computing training loss
+        is_labeled = labels == labels
```

Comment:
In what scenario will there be nan labels?

Reply:
Copied from the author's code, which seems not to be necessary for ogbg-molhiv. Removed it for now. This can be common if the dataset is for multi-label property prediction.

Reply:
It turns out that there are many missing labels in ogbg-molpcba, which is for 128 tasks. I've added it back.